src/mpc-hc/MainFrm.cpp

```diff
@@ -17629,6 +17670,17 @@ void CMainFrame::CloseMedia(bool bNextIsQueued/* = false*/)
         }
     }
 
+    // save external subtitle
+    if (g_bExternalSubtitle &&
+        m_pCurrentSubInput.pSubStream && m_pCurrentSubInput.pSubStream->GetPath().IsEmpty()) {
+        const auto& s = AfxGetAppSettings();
+        if (s.bAutoSaveDownloadedSubtitles) {
+            int pos = 0;
+            const CString& dir = s.strSubtitlePaths.Tokenize(_T(";"), pos);
+            SubtitlesSave(dir, true);

```

Comment:
Also if save was successful, it should update the empty path value with the new location.

File history also needs to be updated, but that could be done in a future patch.